functions/explorers/[slug].ts

```diff
@@ -0,0 +1,150 @@
+import { Env, Etag, extensions } from "../_common/env.js"
+import { extractOptions } from "../_common/imageOptions.js"
+import { buildExplorerProps, Explorer } from "@ourworldindata/explorer"
+import { handlePageNotFound } from "../_common/redirectTools.js"
+import { renderSvgToPng } from "../_common/grapherRenderer.js"
+import { IRequestStrict, Router, error, cors, png } from "itty-router"
+import { Bounds, Url } from "@ourworldindata/utils"
+import {
+    getSelectedEntityNamesParam,
+    migrateSelectedEntityNamesParam,
+    SelectionArray,
+} from "@ourworldindata/grapher"
+import { rewriteMetaTags } from "../_common/grapherTools.js"
+
+const { preflight, corsify } = cors({
+    allowMethods: ["GET", "OPTIONS", "HEAD"],
+})
+
+const router = Router<
+    IRequestStrict,
+    [URL, Env, Etag, EventContext<unknown, any, Record<string, unknown>>]
+>({
+    before: [preflight],
+    finally: [corsify],
+})
+router
+    .get(
+        `/explorers/:slug${extensions.svg}`,
+        async (_, { searchParams }, env) => {
+            console.log("Handling explorer SVG thumbnail request")
+            return handleThumbnailRequest(searchParams, env, "svg")
+        }
+    )
+    .get(
+        `/explorers/:slug${extensions.png}`,
+        async (_, { searchParams }, env) => {
+            console.log("Handling explorer PNG thumbnail request")
+            return handleThumbnailRequest(searchParams, env, "png")
+        }
+    )
+    .get(
+        "/explorers/:slug",
+        async ({ params: { slug } }, { searchParams }, env) => {
+            console.log("Handling explorer HTML page request")
+            return handleHtmlPageRequest(slug, searchParams, env)
+        }
+    )
+    .all("*", () => error(404, "Route not defined"))
+
+async function handleThumbnailRequest(
+    searchParams: URLSearchParams,
+    env: Env,
+    extension: "png" | "svg"
+) {
+    const options = extractOptions(searchParams)
+    const url = env.url
+    url.href = url.href.replace(`.${extension}`, "")
+    const explorerPage = await env.ASSETS.fetch(url, {
+        redirect: "manual",
```

Comment:
TODO for me: check that redirects are still handled correctly.
In particular, check that the redirect `coronavirus-data-explorer -> covid` works.

Reply:
Seems to work :)